### ISSUE TITLE
Constraints warning on running XLFormViewController

### ISSUE THREAD

**Issue by sumowesley** [xmartlabs/XLForm#494]
When I am running an XLFormViewController in my app I get the below warning - I also see the same warning in the example application on the start screen.

>2015-08-07 04:11:37.102 XLForm[70659:41051330] Warning once only: Detected a case where constraints ambiguously suggest a height of zero for a tableview cell's content view. We're considering the collapse unintentional and using standard height instead.

**Comment by conradwt** [comment]
@sumowesley Do you have an example Xcode 6.4 or Xcode 7 project which reproduces the issue?

**Comment by sumowesley** [comment]
Conrad - it's the example app that I downloaded from github - warning appears on the first screen.

**Comment by conradwt** [comment]
@sumowesley Is the Swift or the Objective-C example?  Also, are you using Xcode 6.4 or Xcode 7 beta 5?

**Comment by sumowesley** [comment]
Xcode 6.4 and Objective-C

**Comment by sumowesley** [comment]
Sorry, target is IOS 8.4

**Comment by mkoppanen** [comment]
Reversing this https://github.com/xmartlabs/XLForm/commit/6725d0a9b4a07ccff7e6be36f1525828ac320ca1 seems to remove the error message.

**Comment by sumowesley** [comment]
Left my comment against the commit - wouldn't want #486 back.

**Comment by davidrothera** [comment]
Seeing this also, but only when there is a `XLFormRowDescriptorTypeButton` active in the controller

**Comment by sumowesley** [comment]
Certainly happening on my own projects when I don't have XLFormRowDescriptorTypeButton in the view controller.

**Comment by davidrothera** [comment]
@sumowesley perhaps there is another type that is causing it then as well, I only have Text types apart from the button and it doesn't happen when there are no buttons in the form.

Which types are you using?

**Comment by sumowesley** [comment]
I'm using text, stepper, push selector, info, textview & date inline.

**Comment by staufman** [comment]
I am seeing this as well and it does have a material impact. In my case, I believe the error is causing nondeterministic behavior for my text view cells (the title is the wrong size/font).

Anyone find a solution to this?

**Comment by davidrothera** [comment]
With the warning I was seeing I would be very surprised if it was having an
actual impact on the app as it was more f a warning that it was guessing
about constraints.

On Monday, 28 September 2015, Salil Apte <notifications@github.com> wrote:

> I am seeing this as well and it does have a material impact. In my case, I
> believe the error is causing nondeterministic behavior for my text view
> cells (the title is the wrong size/font).
>
> Anyone find a solution to this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xmartlabs/XLForm/issues/494#issuecomment-143831269>.
>


-- 
David Rothera

**Comment by dbarabander** [comment]
Did anyone figure out an acceptable workaround for this besides reversting a commit ?

**Comment by sumowesley** [comment]
Haven't seen anything yet. Backing out of the commit that fixes #486 would not thrill me at all.

**Comment by mats-claassen** [comment]
This warning goes away if you set some value to `tableView.rowHeight` other than UITableViewAutomaticDimension. I believe this is just a warning and not really breaking anything.

To my understanding this happens with cells that have no Autolayout constraints and therefore the automatically calculated height is 0.

I have lived with this warning for a time now and never noticed problems but a simple workaround is to set:
```
self.tableView.rowHeight = 44 // or whatever
```
In the viewDidLoad method of your view controller.

And for those rows that profit from UITableViewAutomaticDimension you can use
```
row.height = UITableViewAutomaticDimension;
```

But again, I don't think it is a problem.
